### PR TITLE
Backport block editor persisted preferences (refresh)

### PR DESCRIPTION
**Pull request by ironprogrammer**
Refresh of #3219.

- Adds PHPunit test for new `wp_register_persisted_preferences_meta` function.
- Validates registration and structure of meta key.

Trac ticket: https://core.trac.wordpress.org/ticket/56467

Props noisysocks costdev talldanwp

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: tests/phpunit/tests/user/wpRegisterPersistedPreferencesMeta.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+		$this->assertIsArray( $wp_meta_keys, 'No meta keys exist' );
+		$this->assertArrayHasKey(
+			$meta_key,
+			$wp_meta_keys['user'][''],
+			'The expected meta key was not registered'
+		);
+
+		// Test to detect changes in meta key structure.
+		$structure = $wp_meta_keys['user'][''][ $meta_key ];
```
Comment (costdev) [inline tests/phpunit/tests/user/wpRegisterPersistedPreferencesMeta.php]: Hmm, on a second look, should we change this variable name to `$actual` for consistency and clarity?

**Comment by talldan** [comment]
> Props noisysocks costdev talldan

It's talldanwp on wordpress.org. Thanks for adding the tests. 😄

File: tests/phpunit/tests/user/wpRegisterPersistedPreferencesMeta.php (hunk 1 above)
Comment (ockham) [reply tests/phpunit/tests/user/wpRegisterPersistedPreferencesMeta.php]:
```diff
diff --git a/tests/phpunit/tests/user/wpRegisterPersistedPreferencesMeta.php b/tests/phpunit/tests/user/wpRegisterPersistedPreferencesMeta.php
index ddf604eab1..436f373963 100644
--- a/tests/phpunit/tests/user/wpRegisterPersistedPreferencesMeta.php
+++ b/tests/phpunit/tests/user/wpRegisterPersistedPreferencesMeta.php
@@ -28,7 +28,7 @@ class Tests_User_wpRegisterPersistedPreferencesMeta extends WP_UnitTestCase {
                );
 
                // Test to detect changes in meta key structure.
-               $structure = $wp_meta_keys['user'][''][ $meta_key ];
+               $actual = $wp_meta_keys['user'][''][ $meta_key ];
 
                $this->assertSame(
                        array(
@@ -55,7 +55,7 @@ class Tests_User_wpRegisterPersistedPreferencesMeta extends WP_UnitTestCase {
                                        ),
                                ),
                        ),
-                       $structure,
+                       $actual,
                        'The registered metadata did not have the expected structure'
                );
        }
```

**Review by ockham — APPROVED** [review]
LGTM :shipit: 

Could a Core committer please apply the [test variable change](https://github.com/WordPress/wordpress-develop/pull/3253#discussion_r971941982), and apply with (with props to `noisysocks` `costdev` `talldanwp`)? 😊 

cc/ @audrasjb @hellofromtonya

File: tests/phpunit/tests/user/wpRegisterPersistedPreferencesMeta.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
+<?php
+
+/**
+ * @group user
+ *
+ * @covers ::wp_register_persisted_preferences_meta
+ */
+class Tests_User_wpRegisterPersistedPreferencesMeta extends WP_UnitTestCase {
```
Comment (hellofromtonya) [inline tests/phpunit/tests/user/wpRegisterPersistedPreferencesMeta.php:8]:
```suggestion
class Tests_User_WpRegisterPersistedPreferencesMeta extends WP_UnitTestCase {
```

File: tests/phpunit/tests/user/wpRegisterPersistedPreferencesMeta.php (hunk 1 above)
Comment (hellofromtonya) [reply tests/phpunit/tests/user/wpRegisterPersistedPreferencesMeta.php]: Done in https://github.com/WordPress/wordpress-develop/pull/3253/commits/0af827e7c4a11c6be944091d36c18c46201f7a93 (though I removed the variable for consistency ✅

**Comment by hellofromtonya** [comment]
# Test Report

Env:
* OS: macOS Monterey
* Localhost: wp-env
* Plugins: none
* Theme: TT2
* Browsers: Chrome and Firefox

## Tests and Results
All tests will use Chrome and Firefox to check if preferences persist between browsers and on reload in same browser.

### Test 1: Posts UI

Changing preferences in one browser:
* should persist in same browser on page load ✅ 
* should persist in the other browser on page load ✅ 

persist-post

### Test 2: Site Editor

Changing preferences in one browser:
* should persist in same browser on page load ✅ 
* should persist in the other browser on page load ✅ 

persist-site-editor

### Test 3: Should not change other UIs

Changing preferences in Post UI should not change Site Editor preferences ✅ 

persist-screensgif

**Review by hellofromtonya — APPROVED** [review]
Works as expected ✅ 
Meets coding standards ✅ 
Tests pass ✅ 
All concerns addressed ✅ 

Prepping commit

**Comment by hellofromtonya** [comment]
Committed via https://core.trac.wordpress.org/changeset/54182. Thank you everyone for your contributions!

**Pull request closed without merging**